### PR TITLE
Restructuring the entry listing info endpoint

### LINKED ISSUE #70: Extending "reserved keywords"
There may be a need/want to add reserved words. Now, only `info` is a reserved keyword.

> Technically, I think only `info` needs to be reserved in the sense described under `ID`, since the point is to not get collisions between, e.g., /structures/info and /structures/42. If we intend to actually reserve our other endpoint names here, then `extensions` should also be in this list. (In fact, it might be a very good idea to at least also reserve `extensions`, since it facilitates future use cases where someone needs to add endpoint-specific extensions...)

_Originally posted by @rartino in https://github.com/Materials-Consortia/OPTiMaDe/pull/68_

### PR DESCRIPTION
**Pull request by merkys**
Fixes #70

### CODE REVIEW

File: optimade.md [hunk 1]
```diff
@@ -547,6 +546,7 @@ following endpoints:
 * an "entry listing" endpoint
 * a "single entry" endpoint
 * an introspection `info` endpoint
+* an "entry listing" introspection `info` endpoint
```
Comment (CasperWA) [inline optimade.md]: Shouldn't this be "merged" with the line above instead? E.g.:

```suggestion
* a base and an "entry listing" introspection `info` endpoint
```

Or

```suggestion
* a common introspection `info` endpoint
```

Comment (merkys) [reply optimade.md]: Well, we have separate "entry listing endpoint" and "single entry endpoint", which, I believe, are `/<entry_type>` and `/<entry_type>/<id>`, respectively. Thus I'd either merge both, or none. What'd you say?

File: optimade.md [hunk 2]
```diff
@@ -118,7 +118,6 @@ interpreted as described in [RFC 2119](http://tools.ietf.org/html/rfc2119).
   The following constraints and conventions apply to IDs:
   * Taken together, the ID and entry type MUST uniquely identify the entry.
   * IDs MUST be URL-safe strings; in particular, they MUST NOT contain commas.
```
Comment (rartino) [inline optimade.md]: Was not the idea to let this PR also remove the requirement that IDs MUST be URL-safe?

Comment (merkys) [reply optimade.md]: Right. I will address this in a separate PR.

**Comment by merkys** [comment]
I have merged this PR for now despite the ongoing discussion about endpoint list (by @CasperWA) and remaining constraints on IDs (by @rartino). I see both remaining issues as somewhat logically separate from what has been done in this PR. I will open a follow-up PR for clearing up things concerning IDs.

File: optimade.md (hunk 1 above)
Comment (CasperWA) [reply optimade.md]: After writing a long(ish) response here, I have deleted it and can appreciate your point of view now. The reason why is that the client will have to deal differently with an "entry listing" endpoint and a "single entry" endpoint, due to the way the response is set up for the `data` field.
The same is true for the `info` endpoint when dealing with the "base URL" `info` endpoint or the "entry listing" `info` endpoint(s).
So it makes sense to divide them here as well.

**Pull request merged**